bin/vault_sync_kv_credential.py

```diff
@@ -143,36 +158,65 @@ def stream_events(self, inputs, ew):
+        # TODO - this should be .debug(), but level setting doesn't seem to be working?
+        self._logger.info("{0}: latest KV secret version: {1}".format(input_name, fetched_secret_version))
+
+        fetched_vault_username = vault_kv_secret.key(self.vault_username_key)
+        fetched_vault_password = vault_kv_secret.key(self.vault_password_key)
 
         credential_session = self.service
         # switch app context if one was specified
         if self.credential_app:
             credential_session = client.connect(app=self.credential_app, token=self.service.token)
 
         # default realm to empty string
-        credential_title = "{0}:{1}:".format(self.credential_realm or "", self.credential_username)
+        credential_title = "{0}:{1}:".format(self.credential_realm or "", fetched_vault_username)
 
         self._logger.debug("{0}: working with credential: {1}".format(input_name, credential_title))
 
         if credential_title in credential_session.storage_passwords:
             self._logger.debug("{0}: found existing credential".format(input_name))
 
             found_credential = credential_session.storage_passwords[credential_title]
-            if found_credential.content.clear_password != fetched_vault_secret:
+            if found_credential.content.clear_password != fetched_vault_password:
                 self._logger.debug("{0}: stored credential is out of date, updating".format(input_name))
-                found_credential.update(password=fetched_vault_secret)
+                found_credential.update(password=fetched_vault_password)
             else:
                 self._logger.debug("{0}: stored credential is up to date, doing nothing".format(input_name))
 
         else:
             self._logger.info("{0}: no existing credential found, creating".format(input_name))
-            credential_session.storage_passwords.create(fetched_vault_secret, self.credential_username, self.credential_realm)
+            credential_session.storage_passwords.create(fetched_vault_password, fetched_vault_username, self.credential_realm)
             self._logger.debug("{0}: credential created".format(input_name))
 
+        if self.remove_old_versions:
+            # TODO - do type conversions when fetching arguments
+            oldest_removeable_version = vault_kv_secret.version() - int(self.remove_old_versions)
+
+            self._logger.info("{0}: removeable versions".format(input_name))
+            for previous_version in vault_kv_secret.previous_versions():
+                if previous_version.version() < oldest_removeable_version:
+                    break
+                self._logger.info("  {0}: previous version: {1}".format(input_name, previous_version.version()))
+
+                previous_version_vault_username = previous_version.key(self.vault_username_key)
+                previous_version_vault_password = previous_version.key(self.vault_password_key)
```

Comment:
unused variable?

Reply:
Yup.  I initially thought I'd look for a stale username/password combination...then realized that's impossible, because a username (in the same realm and app/user context) can't have different passwords.  Forgot to remove this variable when I made that realization.

Fixed.